meson.build

```diff
@@ -1,12 +1,22 @@
 project('SIMDe', 'c', 'cpp',
         default_options: ['c_std=c99'],
         license: 'MIT',
+        meson_version : '>=0.54.0',
         version: '0.7.2')
 # also update SIMDE_VESION_* in simde/simde-common.h
 
 cc = meson.get_compiler('c')
 cxx = meson.get_compiler('cpp')
 
+foreach additional_arg : [ '-Wno-unused-function', '-Wno-reduced-alignment' ]
+    if cc.has_argument(additional_arg)
+        add_global_arguments(additional_arg, language : 'c')
+    endif
+    if cxx.has_argument(additional_arg)
+        add_global_arguments(additional_arg, language : 'cpp')
+    endif
+endforeach
+
```

Comment:
I get your position, it should be right to let users of SIMDe decide which warnings to turn on and off rather than decide for them. In other words, it's better to find which code actually causes these warnings rather than turn them off explicitly. So...
* `-Wunused-function` triggers at `simde_assert_close_*`, `simde_assert_equal_*`, `simde_test_codegen_random_*`, `simde_test_codegen_write_*`, `simde_test_*_precision_to_slop`, `simde_test_x86_assert_close_*`, `simde_test_x86_assert_equal_*`, `simde_test_x86_random_extract_*`, `simde_test_x86_random_*`, `simde_test_x86_write_*` functions. I guess that's because there are `static` prototypes of them in `test.h`, but not all of them are used in every file which includes it. How might you suggest to fix this? I have no idea where neither stripping it off of `static` nor keeping this `-Wno-unused-function` is involved.
* About `-Wreduced-alignment`, E2K has maximum platform alignment available of 16 bytes. I tried to set `SIMDE_ALIGN_PLATFORM_MAXIMUM 16`, but had no success; probably it's some sort of weird internal stuff of compiler. Still working on figuring out exact combination of pack/align attributes which actually triggers this warning. As long as I've seen, it shows up on various `test_vec[8]` variables in tests. Gotta find a simple code which triggers this, and look for a method to avoid this. Still WIP for now.

Reply:
The unused functions are all in the tests; we actually try to [disable this warning in the tests](https://github.com/simd-everywhere/simde/blob/12069d720f43830ae9791e8b0f4c4fa3c88012a0/test/test.h#L22) already, so it sounds like we just need an lcc implementation of [SIMDE_DIAGNOSTIC_DISABLE_UNUSED_FUNCTION_](https://github.com/simd-everywhere/simde/blob/12069d720f43830ae9791e8b0f4c4fa3c88012a0/simde/simde-diagnostic.h#L232-L240).  Since it has an EDG frontend, I *think* that means it would be in the form of `#pragma diag_suppress ####`.

For reduced alignment, `SIMDE_ALIGN_PLATFORM_MAXIMUM` is the right way to go.  If you want, I can try to take a look at this after I have access; I'm more familiar with the macros and how SIMDe works in general, so I may have an easier time.  This may well be a bug in SIMDe; most compilers don't require a maximum alignment, so it tends to not be as well-tested.

Reply:
Ok, `-Wunused-function` done the right way in e98fb67. For `-Wreduced-alignment`, still under investigation...

Reply:
Finally did the `-Wreduced-alignment` hacks wherever they had to be done in 378b8de, c08e573, e9daa56, and c62dbf1. Considering this resolved finally.